homework/src/index.js

```diff
@@ -1,19 +1,34 @@
 'use strict';
-
 {
-  function fetchJSON(url, cb) {
-    const xhr = new XMLHttpRequest();
-    xhr.open('GET', url);
-    xhr.responseType = 'json';
-    xhr.onload = () => {
-      if (xhr.status < 400) {
-        cb(null, xhr.response);
-      } else {
-        cb(new Error(`Network error: ${xhr.status} - ${xhr.statusText}`));
-      }
-    };
-    xhr.onerror = () => cb(new Error('Network request failed'));
-    xhr.send();
+
+  const root = document.getElementById('root');
+  const header = createAndAppend('header', root, { id: 'header' });
+  const h3 = createAndAppend('h3', header);
+  h3.innerText = 'HYF Repository';
+  const mainDiv = createAndAppend('div', root, { id: 'main' });
+  const repositoryInfo = createAndAppend('div', mainDiv, { id: 'repositoryInfo' });
+  const contributionsInfo = createAndAppend('div', mainDiv, { id: 'contributionsInfo' });
+  const ul = createAndAppend('ul', repositoryInfo, { id: 'list-container' });
+  const Contributions = createAndAppend('h3', contributionsInfo);
+  Contributions.innerText = 'Contributions';
+  const ul2 = createAndAppend('ul', contributionsInfo, { id: 'list-container' });
```

Comment:
Lines 4-14 are potentially executed before the web page is fully loaded. For instance, if you move the `<script>` tag in the HTML file from the `<body>` to the `<head>` tag, you app no longer works, because the browser had not yet created the root `<div>`.

It is best to move this code to the beginning of your `main()` function (just before the `try` statement) to make sure that this code is executed **_after_** the page has full loaded.

Because this code is then no longer global to this file, you will need to pass some variables that are now global as parameters to the function that need them. This is a good thing because functions that receive all the information that they require through the argument list are easier to understand and debug, because they are more or less **_self-contained_**.